Client/Frontend/Browser/FaviconHandler.swift

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import Storage
+import SDWebImage
+import Deferred
+
+class FaviconHandler {
+    static let MaximumFaviconSize = 1 * 1024 * 1024 // 1 MiB file size limit
+
+    private var tabObservers: TabObservers!
+
+    init() {
+        self.tabObservers = registerFor(.didLoadPageMetadata, queue: .main)
```

Comment:
Does this have to be on `.main`?

The alternative would be start creating some generic `NSOperationQueue`s in `TabEventHandler` for (say) `.background`, `.network` etc.